.history/zh/cs-230-recurrent-neural-networks_20191006214850.md

```diff
@@ -0,0 +1,676 @@
+
+&#10230;
+其中Wax,Waa,Wya,ba是相关的系数矩阵, 在时间尺度上被整个网络共享；g1,g2是相关的激活函数。
+<br>
+
+
+**15. The pros and cons of a typical RNN architecture are summed up in the table below:**
+
+&#10230;
+一个典型的RNN体系结构的优点和缺点可概括如下表：
+<br>
+
+
+**16. [Advantages, Possibility of processing input of any length, Model size not increasing with size of input, Computation takes into account historical information, Weights are shared across time]**
+
+&#10230;
+[优点, 可处理任何长度的输入, 模型大小不会随输入大小增加, 计算考虑历史信息, 权重在时间尺度上被整个网络共享]
+<br>
+
+
+**17. [Drawbacks, Computation being slow, Difficulty of accessing information from a long time ago, Cannot consider any future input for the current state]**
+
+&#10230;
+[缺点, 计算缓慢, 难以访问长时间的历史信息, 难以考虑未来时间步的输入对当前状态的影响]
+<br>
+
+
+**18. Applications of RNNs ― RNN models are mostly used in the fields of natural language processing and speech recognition. The different applications are summed up in the table below:**
+
+&#10230;
+RNNs的应用 - RNN模型常用于自然语言处理和语音识别, 下表总结了RNN模型的不同应用场景：
+<br>
+
+
+**19. [Type of RNN, Illustration, Example]**
+
+&#10230;
+[RNN的类型, 图形表示, 示例]
+<br>
+
+
+**20. [One-to-one, One-to-many, Many-to-one, Many-to-many]**
+
+&#10230;
+[一对一, 一对多, 多对一, 多对多]
+<br>
+
+
+**21. [Traditional neural network, Music generation, Sentiment classification, Name entity recognition, Machine translation]**
+
+&#10230;
+[传统神经网络, 音乐生成, 情感分类, 命名实体识别, 机器翻译]
+<br>
+
+
+**22. Loss function ― In the case of a recurrent neural network, the loss function L of all time steps is defined based on the loss at every time step as follows:**
+
+&#10230;
+损失函数 - 在循环神经网络的情况下, 所有时间步长的损失函数L是基于每个时间步长的损失来定义的, 其表示如下：
+<br>
+
+
+**23. Backpropagation through time ― Backpropagation is done at each point in time. At timestep T, the derivative of the loss L with respect to weight matrix W is expressed as follows:**
+
+&#10230;
+随时间反向传播算法(BPTT) - 反向传播在每个时间点完成。在时间步T, 损失函数L相对于权重矩阵W的导数表示如下：
+<br>
+
+
+**24. Handling long term dependencies**
+
+&#10230;
+解决长时间依赖问题
+<br>
+
+
+**25. Commonly used activation functions ― The most common activation functions used in RNN modules are described below:**
+
+&#10230;
+常用的激活函数 - 在RNN模型中常用的激活函数如下所示：
+<br>
+
+
+**26. [Sigmoid, Tanh, RELU]**
+
+&#10230;
+[Sigmoid, Tanh, RELU]
+<br>
+
+
+**27. Vanishing/exploding gradient ― The vanishing and exploding gradient phenomena are often encountered in the context of RNNs. The reason why they happen is that it is difficult to capture long term dependencies because of multiplicative gradient that can be exponentially decreasing/increasing with respect to the number of layers.**
+
+&#10230;
+梯度消失/梯度爆炸 - 梯度消失和梯度爆炸现象常出现在RNN模型中。其原因是该模型结构难以捕获长期依赖性, 因为乘法梯度会随着层数增加而呈指数递减/递增。
+<br>
+
+
+**28. Gradient clipping ― It is a technique used to cope with the exploding gradient problem sometimes encountered when performing backpropagation. By capping the maximum value for the gradient, this phenomenon is controlled in practice.**
+
+&#10230;
+梯度裁剪 - 该方法是用于解决进行反向传播时时而出现梯度爆炸问题的技术。通过限制梯度的最大值, 这种现象在实际中得到了相应的控制。
```

Comment:
To be consistent with the chinese version of book 《deep learning》(lan Goodfellow, et al),**Gradient clipping** will be translated to *梯度截断* rather than **梯度裁剪**.

为与Ian Goodfellow等人编写的《deep learning》中文译本保持一致，将**Gradient clipping**翻译为**梯度截断**而不是**梯度裁剪**。